bot/exts/events/hacktoberfest/_utils.py

```diff
@@ -0,0 +1,452 @@
+    "language:python"                        # in Python.
+)
+
+
+# Util functions for `.hacktoberfest timeleft`
+def in_hacktober() -> bool:
+    """Return True if the current time is within Hacktoberfest."""
+    _, end, start = load_date()
+
+    now = datetime.utcnow()
+
+    return start <= now <= end
+
+
+def load_date() -> tuple[datetime, datetime, datetime]:
+    """Return of a tuple of the current time and the end and start times of the next October."""
+    now = datetime.utcnow()
+    year = now.year
+    if now.month > 10:
+        year += 1
+    end = datetime(year, 11, 1, 12)  # November 1st 12:00 (UTC-12:00)
+    start = datetime(year, 9, 30, 10)  # September 30th 10:00 (UTC+14:00)
+    return now, end, start
+
+
+# Util functions for `.hacktoberfest stats`
+async def get_stats(ctx: commands.Context, github_username: str) -> None:
+    """
+    Query GitHub's API for PRs created by a GitHub user during the month of October.
+
+    PRs with an 'invalid' or 'spam' label are ignored unless merged or approved.
+
+    PRs have to be in a repository that has a 'hacktoberfest' topic,
+    unless the PR is labelled 'hacktoberfest-accepted' for it to count.
+
+    If a valid `github_username` is provided, an embed is generated and posted to the channel.
+
+    Otherwise, a helpful error message is posted.
+    """
+    async with ctx.typing():
+        prs = await get_october_prs(ctx.bot, github_username)
+
+        if prs is None:  # Will be `None` if the user was not found
+            await ctx.send(
+                embed=discord.Embed(
+                    title=random.choice(NEGATIVE_REPLIES),
+                    description=f"GitHub user `{github_username}` was not found.",
+                    colour=discord.Colour.red()
+                )
+            )
+            return
+
+        if prs:
+            stats_embed = await build_stats_embed(ctx.bot, github_username, prs)
+            await ctx.send("Here are some stats!", embed=stats_embed)
+        else:
+            await ctx.send(f"No valid Hacktoberfest PRs found for '{github_username}'")
+
+
+async def build_stats_embed(bot: Bot, github_username: str, prs: list[dict]) -> discord.Embed:
+    """Return a stats embed built from github_username's PRs."""
+    logging.info(f"Building Hacktoberfest embed for GitHub user: '{github_username}'")
+    in_review, accepted = await categorize_prs(bot, prs)
+
+    n = len(accepted) + len(in_review)  # Total number of PRs
+    if n >= PRS_FOR_SHIRT:
+        shirtstr = f"**{github_username} is eligible for hacktoberfest swag!**"
+    elif (remaining_prs := PRS_FOR_SHIRT - n) == 1:
+        shirtstr = f"**{github_username} is 1 PR away from being eligible for hacktoberfest swag!**"
+    else:
+        shirtstr = f"**{github_username} is {remaining_prs} PRs away from being eligible for hacktoberfest swag!**"
+
+    stats_embed = discord.Embed(
+        title=f"{github_username}'s Hacktoberfest",
+        color=Colours.purple,
```

Comment:
This could be made a constant as the hacktoberfest colours are changing every year... or even smarter at the start of hacktoberfest, you can just fetch the GitHub profile picture and extract the most dominant colour.